re2 fails to build on amd64

The package re2 fails to build from source with the following error:

```
  163 |     return Policy::template get_hash_slot_fn<Hash>() == nullptr
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /usr/include/absl/container/internal/raw_hash_set.h:213,
ninja: job failed: /usr/local/bin/c++ -DRE2_USE_ICU -Dre2_EXPORTS -I/home/build -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/re2.dir/re2/prefilter_tree.cc.o -MF CMakeFiles/re2.dir/re2/prefilter_tree.cc.o.d -o CMakeFiles/re2.dir/re2/prefilter_tree.cc.o -c /home/build/re2/prefilter_tree.cc
                 from /usr/include/absl/container/flat_hash_set.h:45,
                 from /home/build/re2/prefilter_tree.h:22,
                 from /home/build/re2/prefilter_tree.cc:5:
/usr/include/absl/container/internal/hash_policy_traits.h: In instantiation of 'static const absl::lts_20240722::container_internal::PolicyFunctions& absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::GetPolicyFunctions() [with Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; Hash = re2::PrefilterTree::PrefilterHash; Eq = re2::PrefilterTree::PrefilterEqual; Alloc = std::allocator<re2::Prefilter*>]':
/usr/include/absl/container/internal/raw_hash_set.h:3825:67:   required from 'std::pair<absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::iterator, bool> absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::find_or_prepare_insert_non_soo(const K&) [with K = re2::Prefilter*; Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; Hash = re2::PrefilterTree::PrefilterHash; Eq = re2::PrefilterTree::PrefilterEqual; Alloc = std::allocator<re2::Prefilter*>]'
 3825 |                                                 GetPolicyFunctions())),
      |                                                 ~~~~~~~~~~~~~~~~~~^~
/usr/include/absl/container/internal/raw_hash_set.h:3887:42:   required from 'std::pair<absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::iterator, bool> absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::find_or_prepare_insert(const K&) [with K = re2::Prefilter*; Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; Hash = re2::PrefilterTree::PrefilterHash; Eq = re2::PrefilterTree::PrefilterEqual; Alloc = std::allocator<re2::Prefilter*>]'
 3887 |     return find_or_prepare_insert_non_soo(key);
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/absl/container/internal/raw_hash_set.h:3453:42:   required from 'std::pair<absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::iterator, bool> absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::EmplaceDecomposable::operator()(const K&, Args&& ...) const [with K = re2::Prefilter*; Args = {re2::Prefilter*&}; Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; Hash = re2::PrefilterTree::PrefilterHash; Eq = re2::PrefilterTree::PrefilterEqual; Alloc = std::allocator<re2::Prefilter*>]'
 3453 |       auto res = s.find_or_prepare_insert(key);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/absl/container/internal/container_memory.h:228:28:   required from 'decltype (declval<F>()(declval<const Arg&>(), declval<Arg>())) absl::lts_20240722::container_internal::DecomposeValue(F&&, Arg&&) [with F = raw_hash_set<FlatHashSetPolicy<re2::Prefilter*>, re2::PrefilterTree::PrefilterHash, re2::PrefilterTree::PrefilterEqual, std::allocator<re2::Prefilter*> >::EmplaceDecomposable; Arg = re2::Prefilter*&; decltype (declval<F>()(declval<const Arg&>(), declval<Arg>())) = std::pair<raw_hash_set<FlatHashSetPolicy<re2::Prefilter*>, re2::PrefilterTree::PrefilterHash, re2::PrefilterTree::PrefilterEqual, std::allocator<re2::Prefilter*> >::iterator, bool>]'
  228 |   return std::forward<F>(f)(key, std::forward<Arg>(arg));
      |          ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/absl/container/flat_hash_set.h:536:52:   required from 'static decltype (absl::lts_20240722::container_internal::DecomposeValue(declval<F>(), (declval<Args>)()...)) absl::lts_20240722::container_internal::FlatHashSetPolicy<T>::apply(F&&, Args&& ...) [with F = absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>, re2::PrefilterTree::PrefilterHash, re2::PrefilterTree::PrefilterEqual, std::allocator<re2::Prefilter*> >::EmplaceDecomposable; Args = {re2::Prefilter*&}; T = re2::Prefilter*; decltype (absl::lts_20240722::container_internal::DecomposeValue(declval<F>(), (declval<Args>)()...)) = std::pair<absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>, re2::PrefilterTree::PrefilterHash, re2::PrefilterTree::PrefilterEqual, std::allocator<re2::Prefilter*> >::iterator, bool>]'
  536 |     return absl::container_internal::DecomposeValue(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  537 |         std::forward<F>(f), std::forward<Args>(args)...);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/absl/container/internal/hash_policy_traits.h:134:20:   required from 'static decltype (P::apply(forward<F>(f), (forward<Ts>)(absl::lts_20240722::container_internal::hash_policy_traits<Policy, <template-parameter-1-2> >::apply::ts)...)) absl::lts_20240722::container_internal::hash_policy_traits<Policy, <template-parameter-1-2> >::apply(F&&, Ts&& ...) [with F = absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>, re2::PrefilterTree::PrefilterHash, re2::PrefilterTree::PrefilterEqual, std::allocator<re2::Prefilter*> >::EmplaceDecomposable; Ts = {re2::Prefilter*&}; P = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; <template-parameter-1-2> = void; decltype (P::apply(forward<F>(f), (forward<Ts>)(absl::lts_20240722::container_internal::hash_policy_traits<Policy, <template-parameter-1-2> >::apply::ts)...)) = std::pair<absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>, re2::PrefilterTree::PrefilterHash, re2::PrefilterTree::PrefilterEqual, std::allocator<re2::Prefilter*> >::iterator, bool>]'
  134 |     return P::apply(std::forward<F>(f), std::forward<Ts>(ts)...);
      |            ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/absl/container/internal/raw_hash_set.h:3012:31:   required from 'std::pair<absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::iterator, bool> absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::emplace(Args&& ...) [with Args = {re2::Prefilter*&}; typename std::enable_if<absl::lts_20240722::container_internal::IsDecomposable<void, absl::lts_20240722::container_internal::hash_policy_traits<Policy>, Hash, Eq, Ts ...>::value, int>::type <anonymous> = 0; Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>; Hash = re2::PrefilterTree::PrefilterHash; Eq = re2::PrefilterTree::PrefilterEqual; Alloc = std::allocator<re2::Prefilter*>]'
 3012 |     return PolicyTraits::apply(EmplaceDecomposable{*this},
      |            ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3013 |                                std::forward<Args>(args)...);
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/build/re2/prefilter_tree.cc:158:21:   required from here
  158 |       nodes->emplace(node);
      |       ~~~~~~~~~~~~~~^~~~~~
/usr/include/absl/container/internal/raw_hash_set.h:4019:56:   in 'constexpr' expansion of 'absl::lts_20240722::container_internal::hash_policy_traits<absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::Prefilter*>, void>::get_hash_slot_fn<re2::PrefilterTree::PrefilterHash>()'
/usr/include/absl/container/internal/hash_policy_traits.h:163:54: error: '(absl::lts_20240722::container_internal::TypeErasedApplyToSlotFn<re2::PrefilterTree::PrefilterHash, re2::Prefilter*> == 0)' is not a constant expression
  163 |     return Policy::template get_hash_slot_fn<Hash>() == nullptr
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /usr/include/absl/container/internal/raw_hash_set.h:213,
                 from /usr/include/absl/container/internal/raw_hash_map.h:26,
ninja: job failed: /usr/local/bin/c++ -DRE2_USE_ICU -Dre2_EXPORTS -I/home/build -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/re2.dir/re2/regexp.cc.o -MF CMakeFiles/re2.dir/re2/regexp.cc.o.d -o CMakeFiles/re2.dir/re2/regexp.cc.o -c /home/build/re2/regexp.cc
                 from /usr/include/absl/container/flat_hash_map.h:45,
                 from /home/build/re2/regexp.cc:20:
/usr/include/absl/container/internal/hash_policy_traits.h: In instantiation of 'static const absl::lts_20240722::container_internal::PolicyFunctions& absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::GetPolicyFunctions() [with Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq; Alloc = std::allocator<std::pair<re2::Regexp* const, int> >]':
/usr/include/absl/container/internal/raw_hash_set.h:3825:67:   required from 'std::pair<absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::iterator, bool> absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::find_or_prepare_insert_non_soo(const K&) [with K = re2::Regexp*; Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq; Alloc = std::allocator<std::pair<re2::Regexp* const, int> >]'
 3825 |                                                 GetPolicyFunctions())),
      |                                                 ~~~~~~~~~~~~~~~~~~^~
/usr/include/absl/container/internal/raw_hash_set.h:3887:42:   required from 'std::pair<absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::iterator, bool> absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::find_or_prepare_insert(const K&) [with K = re2::Regexp*; Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq; Alloc = std::allocator<std::pair<re2::Regexp* const, int> >]'
 3887 |     return find_or_prepare_insert_non_soo(key);
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/absl/container/internal/raw_hash_map.h:212:44:   required from 'std::pair<typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set::iterator, bool> absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::try_emplace_impl(K&&, Args&& ...) [with K = re2::Regexp*; Args = {}; Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq; Alloc = std::allocator<std::pair<re2::Regexp* const, int> >; typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set::iterator = absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq, std::allocator<std::pair<re2::Regexp* const, int> > >::iterator; typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set = absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq, std::allocator<std::pair<re2::Regexp* const, int> > >]'
  212 |     auto res = this->find_or_prepare_insert(k);
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/usr/include/absl/container/internal/raw_hash_map.h:133:28:   required from 'std::pair<typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set::iterator, bool> absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::try_emplace(key_arg<K>&&, Args&& ...) [with K = re2::Regexp*; Args = {}; typename std::enable_if<(! std::is_convertible<K, typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set::const_iterator>::value), int>::type <anonymous> = 0; K* <anonymous> = 0; Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq; Alloc = std::allocator<std::pair<re2::Regexp* const, int> >; typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set::iterator = absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq, std::allocator<std::pair<re2::Regexp* const, int> > >::iterator; typename absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::raw_hash_set = absl::lts_20240722::container_internal::raw_hash_set<absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash, absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq, std::allocator<std::pair<re2::Regexp* const, int> > >; key_arg<K> = re2::Regexp*]'
  133 |     return try_emplace_impl(std::forward<K>(k), std::forward<Args>(args)...);
      |            ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/absl/container/internal/raw_hash_map.h:184:43:   required from 'absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::MappedReference<P> absl::lts_20240722::container_internal::raw_hash_map<Policy, Hash, Eq, Alloc>::operator[](key_arg<K>&&) [with K = re2::Regexp*; P = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; K* <anonymous> = 0; Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Eq; Alloc = std::allocator<std::pair<re2::Regexp* const, int> >; MappedReference<P> = int&; key_arg<K> = re2::Regexp*]'
  184 |         &this->unchecked_deref(try_emplace(std::forward<K>(key)).first));
      |                                ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/home/build/re2/regexp.cc:97:27:   required from here
   97 |   return (*ref_map())[this];
      |                           ^
/usr/include/absl/container/internal/raw_hash_set.h:4019:56:   in 'constexpr' expansion of 'absl::lts_20240722::container_internal::hash_policy_traits<absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::Regexp*, int>, void>::get_hash_slot_fn<absl::lts_20240722::container_internal::HashEq<re2::Regexp*, void>::Hash>()'
/usr/include/absl/container/internal/hash_policy_traits.h:163:54: error: '(absl::lts_20240722::container_internal::TypeErasedApplyToSlotFn<HashEq<re2::Regexp*, void>::Hash, re2::Regexp*> == 0)' is not a constant expression
  163 |     return Policy::template get_hash_slot_fn<Hash>() == nullptr
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /usr/include/absl/container/internal/raw_hash_set.h:213,
                 from /usr/include/absl/container/internal/raw_hash_map.h:26,
                 from /usr/include/absl/container/flat_hash_map.h:45,
                 from /home/build/re2/dfa.cc:39:
ninja: job failed: /usr/local/bin/c++ -DRE2_USE_ICU -Dre2_EXPORTS -I/home/build -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/re2.dir/re2/dfa.cc.o -MF CMakeFiles/re2.dir/re2/dfa.cc.o.d -o CMakeFiles/re2.dir/re2/dfa.cc.o -c /home/build/re2/dfa.cc
/usr/include/absl/container/internal/hash_policy_traits.h: In instantiation of 'static const absl::lts_20240722::container_internal::PolicyFunctions& absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::GetPolicyFunctions() [with Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::DFA::State*>; Hash = re2::DFA::StateHash; Eq = re2::DFA::StateEqual; Alloc = std::allocator<re2::DFA::State*>]':
/usr/include/absl/container/internal/raw_hash_set.h:2892:53:   required from 'void absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::clear() [with Policy = absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::DFA::State*>; Hash = re2::DFA::StateHash; Eq = re2::DFA::StateEqual; Alloc = std::allocator<re2::DFA::State*>]'
 2892 |       ClearBackingArray(common(), GetPolicyFunctions(), /*reuse=*/cap < 128,
      |                                   ~~~~~~~~~~~~~~~~~~^~
/home/build/re2/dfa.cc:814:21:   required from here
  814 |   state_cache_.clear();
      |   ~~~~~~~~~~~~~~~~~~^~
/usr/include/absl/container/internal/raw_hash_set.h:4019:56:   in 'constexpr' expansion of 'absl::lts_20240722::container_internal::hash_policy_traits<absl::lts_20240722::container_internal::FlatHashSetPolicy<re2::DFA::State*>, void>::get_hash_slot_fn<re2::DFA::StateHash>()'
/usr/include/absl/container/internal/hash_policy_traits.h:163:54: error: '(absl::lts_20240722::container_internal::TypeErasedApplyToSlotFn<re2::DFA::StateHash, re2::DFA::State*> == 0)' is not a constant expression
  163 |     return Policy::template get_hash_slot_fn<Hash>() == nullptr
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/usr/include/absl/container/internal/hash_policy_traits.h: In instantiation of 'static const absl::lts_20240722::container_internal::PolicyFunctions& absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::GetPolicyFunctions() [with Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::DFA::State*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::DFA::State*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::DFA::State*, void>::Eq; Alloc = std::allocator<std::pair<re2::DFA::State* const, int> >]':
/usr/include/absl/container/internal/raw_hash_set.h:2892:53:   required from 'void absl::lts_20240722::container_internal::raw_hash_set<Policy, Hash, Eq, Alloc>::clear() [with Policy = absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::DFA::State*, int>; Hash = absl::lts_20240722::container_internal::HashEq<re2::DFA::State*, void>::Hash; Eq = absl::lts_20240722::container_internal::HashEq<re2::DFA::State*, void>::Eq; Alloc = std::allocator<std::pair<re2::DFA::State* const, int> >]'
 2892 |       ClearBackingArray(common(), GetPolicyFunctions(), /*reuse=*/cap < 128,
      |                                   ~~~~~~~~~~~~~~~~~~^~
/home/build/re2/dfa.cc:2082:34:   required from here
 2082 |   previously_visited_states.clear();
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/usr/include/absl/container/internal/raw_hash_set.h:4019:56:   in 'constexpr' expansion of 'absl::lts_20240722::container_internal::hash_policy_traits<absl::lts_20240722::container_internal::FlatHashMapPolicy<re2::DFA::State*, int>, void>::get_hash_slot_fn<absl::lts_20240722::container_internal::HashEq<re2::DFA::State*, void>::Hash>()'
/usr/include/absl/container/internal/hash_policy_traits.h:163:54: error: '(absl::lts_20240722::container_internal::TypeErasedApplyToSlotFn<HashEq<re2::DFA::State*, void>::Hash, re2::DFA::State*> == 0)' is not a constant expression
  163 |     return Policy::template get_hash_slot_fn<Hash>() == nullptr
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
ninja: subcommands failed
deleting guest dir /tmp/melange-guest-1883763617
deleting workspace dir /tmp/melange-workspace-644447899
removing image path /tmp/melange-guest-3752132299
failed to build package: unable to run package re2 pipeline: unable to run pipeline: unable to run pipeline: exit status 1
```

--- a/re2.yaml
+++ b/re2.yaml
@@ -1,7 +1,7 @@
 package:
   name: re2
-  version: "2024.02.01" # Please enable the pkgconf test on update
-  epoch: 2
+  version: "2024.02.01" # On update, please check if -fdelete-null-pointer-checks is still required
+  epoch: 3
   description: Efficient, principled regular expression library
   copyright:
     - license: BSD-3-Clause
@@ -22,6 +22,9 @@ environment:
       - busybox
       - ca-certificates-bundle
       - icu-dev
+  environment:
+    # https://github.com/wolfi-dev/os/issues/34075
+    CMAKE_CXX_FLAGS: -fdelete-null-pointer-checks
 
 pipeline:
   - uses: fetch
@@ -33,6 +36,7 @@ pipeline:
     with:
       opts: |
         -DBUILD_SHARED_LIBS=ON \
+        -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS" \
         -DRE2_USE_ICU=ON \
         -DRE2_BUILD_TESTING="OFF"
 
@@ -43,9 +47,6 @@ pipeline:
   - uses: strip
 
 subpackages:
-  #   test:
-  #     pipeline:
-  #       - uses: test/pkgconf:
   - name: re2-dev
     pipeline:
       - uses: split/dev
@@ -54,6 +55,9 @@ subpackages:
       runtime:
         - abseil-cpp-dev
         - icu-dev
+    test:
+      pipeline:
+        - uses: test/pkgconf
 
 update:
   enabled: true

